pkg/loki/config_handler.go

```diff
@@ -138,7 +163,15 @@ func (t *Loki) tenantLimitsHandler() func(http.ResponseWriter, *http.Request) {
 			}
 		}
 
-		writeYAMLResponse(w, limit)
+		// Apply allowlist filtering if configured
+		allowlist := t.Cfg.LimitsConfig.TenantLimitsAllowPublish
```

Comment:
This is not using the tenant-scoped limit. So if we override the per-tenant  `TenantLimitsAllowPublish` it will take no effect.

Should we use `limit.TenantLimitsAllowPublish` instead? `limit` comes from`t.TenantLimits.TenantLimits(user)` or  `t.Overrides.DefaultLimits()` if that's empty.